src/src/ComputationCoordinatorImp.java

```diff
@@ -1,52 +1,74 @@
-  private int multiply;
-  private int columns;
-  private boolean valid = false;
-  private int outputOrCompute;
-
-  public ComputationCoordinatorImp(DataStorageImp dataStore, ComputeEngineImp compute, int inputType, String inputFileName, int numberOfMatrices, int rows, int columns, int outputType, String outputFileName, int outputOrComp) {
-    this.dataStore = dataStore;
-    this.computeEngine = compute;
-    this.inputType = inputType;
-    this.inputFileName = inputFileName;
-    this.numberOfMatrices = numberOfMatrices;
-    this.rows = rows;
-    this.columns = columns;
-    this.outputOrCompute = outputOrComp;
-  }
+  private static final int THREAD_POOL_SIZE = 20; // Number of threads to use in the pool
+  private final ExecutorService executorService = Executors.newFixedThreadPool(THREAD_POOL_SIZE);
 
   public ComputationCoordinatorImp() {}
 
   public ComputeResult beginComputation(int inputType, String inputFileName, int numberOfMatrices, int rows, int columns, int outputType, String outputFileName, int outputOrComp) throws IOException {
-    try {
-      this.outputOrCompute = outputOrComp;
-      ComputeRequest compute = new ComputeRequest(inputType, inputFileName, numberOfMatrices, rows, columns, outputType, outputFileName);
-      DataStorageImp dataStorage = new DataStorageImp(compute, this.outputOrCompute);
-      ComputeEngineImp computeEng = new ComputeEngineImp(dataStorage);
-
-      return ComputeResult.SUCCESS;
-    } catch (Exception e) {
-      e.printStackTrace();
-      //throw new IOException("An error occurred during computation.", e);
-      return new ComputeResultImpl(ComputeResultStatus.FAILURE, "An error occurred during computation.");
+    // Submit fixed number of tasks to the thread pool
+    for (int i = 0; i < THREAD_POOL_SIZE; i++) { // Create THREAD_POOL_SIZE threads
+      final int threadIndex = i; // Capture the current index for use inside the lambda
+
+      executorService.submit(() -> {
+        String threadName = Thread.currentThread().getName();
+        System.out.println("Executing computation on thread: " + threadName);
+
+        try {
+          // Create a new ComputeRequest for this thread
+          ComputeRequest compute = new ComputeRequest(inputType, inputFileName, numberOfMatrices, rows, columns, outputType, outputFileName, outputOrComp);
+          DataStorageImp dataStorage = new DataStorageImp(compute, outputOrComp);
+
+          // Generate the unique output file name for this thread
+          String uniqueOutputFileName = generateUniqueFileName(outputFileName, threadIndex);
+          ComputeEngineImp computeEng = new ComputeEngineImp(dataStorage);
+
+          // Call the method to perform the computation with the unique output file name
+          computeEng.multiplyMatrix(dataStorage.getMatrices());
+          dataStorage.writeOutput(uniqueOutputFileName, ";"); // Write output to the unique file
+
+        } catch (Exception e) {
+          e.printStackTrace();
```

Comment:
this does not need to throw an exception, istead it should be ComputeResultImpl(ComputeResultStatus.FAILURE, "An error occurred during computation.");